hare/interfaces.go

```diff
@@ -12,6 +12,10 @@ type layerPatrol interface {
 	SetHareInCharge(types.LayerID)
 }
 
+type blockFetcher interface {
+	GetBlocks(context.Context, []types.BlockID) error
+}
```

Comment:
maybe reuse interface definition? there is really no reason to define it everywhere.

there is an interface for Fetcher in `system` module, GetBlocks can be decoupled from it and used where necessary

Reply:
minor issues (that always make me indecisive)
- i only want one of the interface. so in case i cannot use gomock, i need to create all its empty methods. like tests in cmd/xxx
- i don't know whether we should create mock in the importing pkg or the imported pkg. i feel there is more flexibility creating an interface and mock object at the importing pkg. in this case, instead of creating mock at `system/mocks` i chose to do it in `hare/mocks` with just 1 interface method. of course like you said, there'll be code dupes.

Reply:
> i only want one of the interface. so in case i cannot use gomock, i need to create all its empty methods. like tests in cmd/xxx

i meant that you can decouple 

```go
type BlockFetcher interface {
  FetchBlocks
}
type Fetcher interface {
  BlockFetcher
  AtxFetcher
}
```

and use only a specific subset, like in https://github.com/spacemeshos/go-spacemesh/blob/develop/p2p/pubsub/pubsub.go#L55-L69

> i don't know whether we should create mock in the importing pkg or the imported pkg. i feel there is more flexibility creating an interface and mock object at the importing pkg. in this case, instead of creating mock at system/mocks i chose to do it in hare/mocks with just 1 interface method. of course like you said, there'll be code dupes.

i think mocks should be generated where the interface is defined. but i don't see what flexibility is missing